### PR TITLE
Update Cub Cache Creation to new Method

### PR DESCRIPTION
**Pull request by tylera-nvidia**
update all the code in cub.cu to use the new method of creating cub plans for the cache

### CODE REVIEW

File: include/matx/transforms/cub.h [hunk 1]
```diff
@@ -1222,25 +1221,25 @@ void cub_sum(OutputTensor &a_out, const InputOperator &a,
...
-    auto type =
-        static_cast<detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_SUM> *>(ret.value());
-    type->ExecSum(a_out, a, stream);
-  }
+  using cache_val_type = detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_SUM, int>;
+  detail::GetCache().LookupAndExec<detail::cub_cache_t>(
+      detail::GetCacheIdFromType<detail::cub_cache_t>(),
+      params,
+      [&]() {
+        int test; ///\todo TYLER_TODO: this should not be needed
```
Comment (cliffburdick) [inline include/matx/transforms/cub.h:1233]: What is this for?

File: include/matx/transforms/cub.h [hunk 2]
```diff
@@ -1272,21 +1271,18 @@ void cub_min(OutputTensor &a_out, const InputOperator &a,
...
-        static_cast<detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_MIN> *>(
-            ret.value());
-    type->ExecMin(a_out, a, stream);
-  }
+  using cache_val_type = detail::matxCubPlan_t<OutputTensor, InputOperator, detail::CUB_OP_REDUCE_MIN, int>;
+  detail::GetCache().LookupAndExec<detail::cub_cache_t>(
+      detail::GetCacheIdFromType<detail::cub_cache_t>(),
+      params,
+      [&]() {
+        int test; ///\todo TYLER_TODO: this should not be needed
```
Comment (cliffburdick) [inline include/matx/transforms/cub.h:1279]: Same

**Review by cliffburdick — CHANGES_REQUESTED** [review]

File: include/matx/transforms/cub.h (hunk 2 above)
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h:1279]: I have another comment in this PR explaining why I needed this and asking for a second opinion on how to get around it. I'm not sure how to link to another comment in the same PR... it's at line [1276](https://github.com/NVIDIA/MatX/pull/694/files/1969207f7a78218fe53226b6d3cc0f211d68e2be#diff-7c667b17610a2198f407d07c43ae3ed0aa7f9d9777c0bd7d084df613b5d9d441)

Comment (tylera-nvidia) [reply include/matx/transforms/cub.h:1279]: (I see now they were pending and I forgot to submit them).

File: include/matx/transforms/cub.h (hunk 1 above)
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

File: include/matx/transforms/cub.h (hunk 2 above)
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

[the remark this answers is not in the document]
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

**Comment by tylera-nvidia** [comment]
@cliffburdick  I've updated the plan creations per our discussion yesterday. can you review/approve this PR when you get a chance?

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**